File: common/src/main/java/dev/qixils/aviary/common/Aviary.java
package dev.qixils.aviary.common;

import dev.qixils.aviary.common.chat.Chat;
import dev.qixils.aviary.common.messaging.PluginChannel;
import org.jetbrains.annotations.NotNull;

import java.util.UUID;

/**
 * The core of the Aviary API.
 */
public interface Aviary {

	/**
	 * The namespace used in Aviary's {@link net.kyori.adventure.key.Key keys}.
	 */
	@NotNull String NAMESPACE = "aviary";

	/**
	 * Gets the {@link PluginChannel} for the given name and for the given player's server.
	 *
	 * @param name   the name of the channel
	 * @param player the UUID of the player
	 * @return the channel
	 */
	@NotNull PluginChannel getPluginChannel(@NotNull String name, @NotNull UUID player);

	/**
	 * Gets the {@link Chat} instance for managing chat channels.
	 *
	 * @return the chat instance
	 */
	@NotNull Chat getChat();

	// TODO: registry stuff? (assuming i ever use the damn thing)
}

File: common/src/main/kotlin/dev/qixils/aviary/common/Aviary.kt
package dev.qixils.aviary.common

import dev.qixils.aviary.common.chat.Chat
import dev.qixils.aviary.common.messaging.PluginChannel
import java.util.*

/**
 * The core of the Aviary API.
 */
interface Aviary {

	/**
	 * Gets the [PluginChannel] for the given name and for the given player's server.
	 *
	 * @param name   the name of the channel
	 * @param player the UUID of the player
	 * @return the channel
	 */
	fun pluginChannel(name: String, player: UUID): PluginChannel

	/**
	 * The [Chat] instance for managing chat channels.
	 */
	val chat: Chat

	// TODO: registry stuff? (assuming I ever use the damn thing)

	companion object {

		/**
		 * The namespace used in Aviary's [keys][net.kyori.adventure.key.Key].
		 */
		const val NAMESPACE = "aviary"
	}
}
